add docs for Enterprise Oracle database plugin with instant client table (#30106)
hashicorp/vault: website/content/docs/secrets/databases/oracle.mdx

Oracle is one of the supported plugins for the database secrets engine. It is capable of dynamically generating
credentials based on configured roles for Oracle databases. It also supports [static roles](/vault/docs/secrets/databases#static-roles).

-> **Note:** Enterprise Oracle database secrets engine is available starting with version 0.11.0.
See [Plugin management for Enterprise plugins](/vault/docs/plugins/plugin-management#enterprise-plugins)
for details on Vault Enterprise compatible versions and how to register Enterprise plugins.

> 

Reviewer: I'm not sure what we're trying to say here. Are all versions of the plugin at/after 0.11.0 enterprise only?

Author: @schavis Yes.

Oracle is one of the supported plugins for the database secrets engine. It is capable of dynamically generating
credentials based on configured roles for Oracle databases. It also supports [static roles](/vault/docs/secrets/databases#static-roles).

<Note>

> As of version 0.11.0, the Oracle database secrets plugin is only available as an enterprise plugin.

Reviewer: ```suggestion
The Oracle database secrets plugin is available as an enterprise plugin starting with version 0.11.0.
```

Reviewer 2: Hmm revisiting this section and reading both the original and the suggestion: Do we want to be careful about phrasing that suggests we'll only release the Oracle plugin as enterprise going forward (implying no more CE development and releases)? 

I wonder if that can cause issues, since Product hasn't committed to the stance that we'll never develop or release CE versions of the Oracle plugin again, even though that's the direction it _seems_ like we're headed.

Author: @schavis @helenfufu 
```
Enterprise Oracle database secrets engine is available starting with version 0.11.0.
```
I'm trying to compare to my original version. I'm ok as long as Sarah's susggsion is 100% equivalent but has better word choice. I try to avoid saying anything about Oracle being only Enterprise.

Reviewer 2: I think your latest suggestion is good @thyton! I was just about to suggest this very non-commital statement:
```suggestion
The first enterprise version of the Oracle database secrets plugin is 0.11.0.
```
which has the same exact meaning as your
```
Enterprise Oracle database secrets engine is available starting with version 0.11.0.
```

<Tabs>
<Tab heading="Vault" group="vault">

The Oracle database plugin is not bundled in the core Vault code tree.

For linux/amd64, pre-built binaries can be found at [the releases page](https://releases.hashicorp.com/vault-plugin-database-oracle)

Before running the plugin you will need to have the Oracle Instant Client
library installed. These can be downloaded from Oracle. The libraries will need to
be placed in the default library search path or defined in the ld.so.conf configuration files.

For the Community Oracle database plugin, the version of the Oracle Instant Client SDK the
plugin was built with can be found at its own git repository here: [hashicorp/vault-plugin-database-oracle](https://github.com/hashicorp/vault-plugin-database-oracle)

For the Enterprise Oracle database plugin, see the following table:

> | Plugin Release | Instant Client Version |

Reviewer: ```suggestion

| Plugin release | Instant client version |
```

Include whitespace before tables to keep the markdown parser from getting confused. Also, unless "Instant Client" is the proper name of another company's product, it shouldn't be capitalized.

Author: It looks "Instant Client" is a Oracle product name https://www.oracle.com/ca-en/database/technologies/instant-client/downloads.html
